src/inputs/input/input.jsx

```diff
@@ -362,8 +381,11 @@ class Input extends React.PureComponent {
         if (isFunction(onChange)) {
             onChange(value);
         } else {
-            // eslint-disable-next-line no-underscore-dangle
-            this._input.value = value;
+            if (mask) {
```

Comment:
looks like this should be moved up to the else line (ie, it s/b an else-if)

Reply:
Maybe, but sometimes keeping the nesting helps readability.

i.e. there are two completely independent things going on here:
1) Is it a "controlled component" or not? (outer `if`/`else`)
2) Are we using an input mask or not (inner `if`/`else` within the outer `else` [not a controlled component])

I guess we could do this to refactor/simplify as you suggest and assuage my readability concerns (and I suspect @cbrewer1689's concerns as well):
```JavaScript
if (isFunction(onChange)) { // if it's a controlled component
    onChange(value);
} else if (mask) { // not a controlled component; using an input mask
    this.input.current.inputElement.value = value;
} else { // not a controlled component; not using an input mask
    this.input.current.value = value;
}
```

Reply:
I like the suggestion here.

Reply:
Fixed.